chpl/chpl-service/src/main/java/gov/healthit/chpl/certifiedproduct/service/CertificationStatusEventsService.java

```diff
@@ -41,6 +41,24 @@ public CertificationStatusEventDTO getInitialCertificationEvent(Long listingId)
         return certStatusEventDao.findInitialCertificationEventForCertifiedProduct(listingId);
     }
 
+    public CertificationStatusEvent getCurrentCertificationStatusEvent(Long certifiedProductId) throws EntityRetrievalException {
+        List<CertificationStatusEvent> statusEvents = certStatusEventDao.findByCertifiedProductId(certifiedProductId).stream()
+                .map(dto -> createCertificationStatusEventBasedOnDto(dto))
+                .collect(Collectors.toCollection(ArrayList::new));
+
+        if (statusEvents == null || statusEvents.size() == 0) {
+            return null;
+        }
+
+        CertificationStatusEvent newest = statusEvents.get(0);
+        for (CertificationStatusEvent event : statusEvents) {
+            if (event.getEventDate() > newest.getEventDate()) {
+                newest = event;
+            }
+        }
```

Comment:
Why not do all of this in a streamed `.sort()` in the stream above, then get the first?